src/Tasks/Microsoft.NET.Build.Tasks/FrameworkPackages/FrameworkPackages.cs

```diff
@@ -0,0 +1,180 @@
+    {
+        foreach (var frameworkPackages in toRegister)
+        {
+            if (!FrameworkPackagesByFramework.TryGetValue(frameworkPackages.Framework, out var frameworkPackagesForVersion))
+            {
+                FrameworkPackagesByFramework[frameworkPackages.Framework] = frameworkPackagesForVersion = [];
+            }
+
+            var frameworkKey = GetFrameworkKey(frameworkPackages.FrameworkName);
+            frameworkPackagesForVersion[frameworkKey] = frameworkPackages;
+        }
+    }
+
+    public static FrameworkPackages[] GetFrameworkPackages(NuGetFramework framework, string[] frameworkReferences)
+    {
+        var frameworkPackages = new List<FrameworkPackages>();
+
+        if (frameworkReferences.Length == 0)
+        {
+            frameworkReferences = [DefaultFrameworkKey];
+        }
+
+        foreach (var frameworkReference in frameworkReferences)
+        {
+            var frameworkKey = GetFrameworkKey(frameworkReference);
+            if (FrameworkPackagesByFramework.TryGetValue(framework, out var frameworkPackagesForVersion) &&
+                frameworkPackagesForVersion.TryGetValue(frameworkKey, out var frameworkPackage))
+            {
+                frameworkPackages.Add(frameworkPackage);
+            }
+            else
+            {
+                // if we didn't predefine the package overrides, load them from the targeting pack
+                // we might just leave this out since in future frameworks we'll have this functionality built into NuGet.Frameworks
+                var frameworkPackagesFromPack = LoadFrameworkPackagesFromPack(framework, frameworkReference) ?? new FrameworkPackages(framework, frameworkReference);
+
+                Register(frameworkPackagesFromPack);
+
+                frameworkPackages.Add(frameworkPackagesFromPack);
+            }
+        }
+
+        return frameworkPackages.ToArray();
+    }
+
+    private static FrameworkPackages LoadFrameworkPackagesFromPack(NuGetFramework framework, string frameworkName)
+    {
+        if (framework is null || framework.Framework != FrameworkConstants.FrameworkIdentifiers.NetCoreApp)
+        {
+            return null;
+        }
+
+        var reducer = new FrameworkReducer();
+        var frameworkKey = GetFrameworkKey(frameworkName);
+        var candidateFrameworks = FrameworkPackagesByFramework.Where(pair => pair.Value.ContainsKey(frameworkKey)).Select(pair => pair.Key);
+        var nearestFramework = reducer.GetNearest(framework, candidateFrameworks);
+
+        var frameworkPackages = nearestFramework is null ?
+            new FrameworkPackages(framework, frameworkName) :
+            new FrameworkPackages(framework, frameworkName, FrameworkPackagesByFramework[nearestFramework][frameworkKey]);
+
+        // packs location : %ProgramFiles%\dotnet\packs
+        var packsFolder = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "dotnet", "packs", frameworkName + ".Ref");
```

Comment:
The SDK should have a better way to do this.  Can't you pass in the location that the SDK will probe?

Reply:
And what happens with an x86 SDK on x64 or x64 on arm64?

Reply:
This has been fixed to pass in the targeting pack root.  I hadn't paid much attention to the code I copied in so I didn't realize it was doing this.